tests/asgi/tests.py

```diff
@@ -235,11 +235,35 @@ def __call__(self, **kwargs):
         # Give response.close() time to finish.
         await communicator.wait()
 
-        # At this point, AsyncToSync does not have a current executor. Thus
-        # SyncToAsync falls-back to .single_thread_executor.
-        target_thread = next(iter(SyncToAsync.single_thread_executor._threads))
+        # At this point, AsyncToSync should have executed the signals in the
+        # same thread.
         request_started_thread, request_finished_thread = signal_handler.threads
-        self.assertEqual(request_started_thread, target_thread)
-        self.assertEqual(request_finished_thread, target_thread)
+        self.assertEqual(request_started_thread, request_finished_thread)
         request_started.disconnect(signal_handler)
         request_finished.disconnect(signal_handler)
+
+    async def test_concurrent_async_uses_multiple_thread_pools(self):
+        # Perform a basic request.
+        application = get_asgi_application()
+        scope = self.async_request_factory._base_scope(path='/wait/')
+
+        communicators = []
+        for _ in range(2):
+            # Send 2 requests, which will be run concurrently
+            communicators.append(ApplicationCommunicator(application, scope))
+            await communicators[-1].send_input({'type': 'http.request'})
+
+        # Verify the request completes
```

Comment:
To add more explanation, comments should state the expected behavior and omit prefixes like "Ensure that"/"Verify" since all tests verify things. This guideline is from [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).

Reply:
Thanks for the feedback! Comments updated 😄